src/main/java/core/basesyntax/service/ActionStrategyImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.action.ActionHandler;
+import java.util.Map;
+
+public class ActionStrategyImpl implements ActionStrategy {
+    private Map<Operation, ActionHandler> actionHandlerMap;
+
+    public ActionStrategyImpl(Map<Operation, ActionHandler> actionHandlerMap) {
+        this.actionHandlerMap = actionHandlerMap;
+    }
+
+    public ActionStrategyImpl() {
+    }
```

Comment:
redundant constructor

Reply:
it is not redaunt, because I need to create object of this class in class ShopServiceImpl without dropping map there